src/ServiceManagement/StorSimple/Commands.StorSimple/Cmdlets/BackupPolicy/NewAzureStorSimpleDeviceBackupScheduleAddConfig.cs

```diff
@@ -0,0 +1,104 @@
+    public class NewAzureStorSimpleDeviceBackupScheduleAddConfig : StorSimpleCmdletBase 
+    {
+        [Parameter(Position = 0, Mandatory = true, HelpMessage = StorSimpleCmdletHelpMessage.HelpMessageBackupTypeDesc)]
+        [ValidateNotNullOrEmptyAttribute]
+        [ValidateSet("LocalSnapshot", "CloudSnapshot")]
+        public String BackupType { get; set; }
+
+        [Parameter(Position = 1, Mandatory = true, HelpMessage = StorSimpleCmdletHelpMessage.HelpMessageRecurrenceTypeDesc)]
+        [ValidateNotNullOrEmptyAttribute]
+        [ValidateSet("Minutes", "Hourly", "Daily", "Weekly")]
+        public String RecurrenceType { get; set; }
+
+        [Parameter(Position = 2, Mandatory = true, HelpMessage = StorSimpleCmdletHelpMessage.HelpMessageRecurrenceValueDesc)]
+        [ValidateNotNullOrEmptyAttribute]
+        public int RecurrenceValue { get; set; }
+
+        [Parameter(Position = 3, Mandatory = true, HelpMessage = StorSimpleCmdletHelpMessage.HelpMessageRetentionCountDesc)]
+        [ValidateNotNullOrEmptyAttribute]
+        public long RetentionCount { get; set; }
+
+        [Parameter(Position = 4, Mandatory = false, HelpMessage = StorSimpleCmdletHelpMessage.HelpMessageBackupStartFromDesc)]
+        public String StartFromDateTime { get; set; }
+
+        [Parameter(Position = 5, Mandatory = true, HelpMessage = StorSimpleCmdletHelpMessage.HelpMessageBackupEnabledDesc)]
+        public bool Enabled { get; set; }
+
+        private ScheduleStatus scheduleStatus;
+        private DateTime StartFromDt;
+
+        private void ProcessParameters()
+        {
+            if (String.IsNullOrEmpty(StartFromDateTime))
+            {
+                StartFromDt = DateTime.Now;
+            }
+            else
+            {
+                bool dateTimeValid = DateTime.TryParse(StartFromDateTime, out StartFromDt); 
+                
+                if (!dateTimeValid)
+                {
+                    throw new ArgumentException(Resources.StartFromDateForBackupNotValid);
+                }
+            }
+            scheduleStatus = Enabled ? ScheduleStatus.Enabled : ScheduleStatus.Disabled;
+            if (BackupType == "Invalid")
+            {
+                throw new ArgumentException("BackupType cannot be Invalid");
+            }
+
+            if (RetentionCount < 1 || RetentionCount > 64)
+            {
+                throw new ArgumentException("RetentionCount value should be 1 - 64");
+            }
+
+            if (RecurrenceType == "Invalid")
+            {
+                throw new ArgumentException("RecurrenceType cannot be Invalid");
+            }
+
+            if (RecurrenceValue <= 0)
+            {
+                throw new ArgumentException("RecurrenceValue should be >=0");
+            }
+        }
+
+        public override void ExecuteCmdlet()
+        {
+            try
+            {
```

Comment:
We can solve this by having a virtual method void CmdletCore() in cmdlet base, which will be overriden in every cmdlet whereas the base class will call this method in a try-catch block. However this seems to be P2, since I see that Commands.RecoveryServices has their code checked in with the current design. We plan to fix this in the later v2 release, please let me know your thoughts.